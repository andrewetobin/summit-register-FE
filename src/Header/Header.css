.Header {     
  align-items: center;
  background: rgba(165, 165, 165, 0.68);
  display: flex;
  justify-content: space-around;
  padding-top: 12px;
  width: 80%;
}

.main-title {
  color: white;
  font-size: 30px;
}
  
.logo {
  height: 60px;
}

.back-btn {
  background: transparent;
  border: none;
  display: none;
  font-size: 22px;
  margin: 70px 0;
  padding: 5px;
  transition: all 0.4s ease-in-out;
}

.back-btn:hover {
  background: black;
  color: white;
  transition: all 0.4s ease-in-out;
}