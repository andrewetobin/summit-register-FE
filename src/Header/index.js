import React, { Component } from "react";
import logo from "../utilities/Images/summit-2.png";
import PropTypes from "prop-types";
import "./Header.css";

class Header extends Component {
  render() {
    return (
      <header className="Header">
        <button className="back-btn" onClick={this.props.handleBackButton}>
          Back
        </button>
        <h1 className="main-title">Summit Register</h1>
        <img className="logo" alt="logo" src={logo} />
      </header>
    );
  }
}

Header.propTypes = {};

export default Header;
