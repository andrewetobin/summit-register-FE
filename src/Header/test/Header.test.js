import React from "react";
import ReactDOM from "react-dom";
import Header from "..";
import { shallow } from "enzyme";

describe("Header", () => {
  let wrapper;
  const currentMountainData = {
    attributes: {
      altitude: 14265,
      difficulty: "Blue",
      name: "Castle Peak",
      range: "Elk",
      registries: { data: [] },
      summit: "39.009647,-106.86144"
    },
    id: "12",
    type: "mountain"
  };

  beforeEach(() => {
    wrapper = shallow(
      <Header
        currentDisplay={"start"}
        handleBackButton={jest.fn()}
        currentMountainData={currentMountainData}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Header
        currentDisplay={"start"}
        handleBackButton={jest.fn()}
        currentMountainData={currentMountainData}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("display text", () => {
    it("should display 'summit register' on start screen", () => {
      const displayText = wrapper.find(".main-title");
      expect(displayText.text()).toBe("Summit Register");
    });

    it("should display mountain name if not on start screen", () => {
      wrapper = shallow(
        <Header
          currentDisplay={"info"}
          handleBackButton={jest.fn()}
          currentMountainData={currentMountainData}
        />
      );
      const displayText = wrapper.find(".main-title");
      expect(displayText.text()).toBe("Castle Peak");
    });
  });
});
