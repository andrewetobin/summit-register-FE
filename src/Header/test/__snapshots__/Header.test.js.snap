// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentDisplay="start"
    currentMountainData={
      Object {
        "attributes": Object {
          "altitude": 14265,
          "difficulty": "Blue",
          "name": "Castle Peak",
          "range": "Elk",
          "registries": Object {
            "data": Array [],
          },
          "summit": "39.009647,-106.86144",
        },
        "id": "12",
        "type": "mountain",
      }
    }
    handleBackButton={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h1
          className="main-title"
        >
          Summit Register
        </h1>,
        <img
          alt="logo"
          className="logo"
          src="summit-2.png"
        />,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Summit Register",
          "className": "main-title",
        },
        "ref": null,
        "rendered": "Summit Register",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "logo",
          "src": "summit-2.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h1
            className="main-title"
          >
            Summit Register
          </h1>,
          <img
            alt="logo"
            className="logo"
            src="summit-2.png"
          />,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Summit Register",
            "className": "main-title",
          },
          "ref": null,
          "rendered": "Summit Register",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "logo",
            "src": "summit-2.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
